### ISSUE TITLE
Merge `explore users` with `explore orgs`

### ISSUE THREAD

**Issue by delvh** [go-gitea/gitea#29803]
### Feature Description

I have never understood why Gitea splits the explore page into three parts:
`repos, users, orgs`.
Wouldn't it suffice to have `repos, users` (and `orgs` being a part of `users`)?
(Alternatively, the distinction could also be `repos, owner`, however I don't think that's any better either as owner is only a code-internal concept we haven't communicated to our end users, and orgs are on all accounts users too)

In the end, if a user searches for another user, they likely don't know if they are explicitly looking for a user or an organization, they most likely only know the username they are looking for.
As such, it would be more user-friendly to merge the two pages into one that shows both users and orgs simultaneously.

If there is interest, we can still add a toggle later on, that when activated only searches for users XOR orgs, but not both.
But as a first step, I'd suggest merging both explore pages together and getting rid of the (unnecessary) distinction.

**Comment by denyskon** [comment]
The underlying code and templates are absolutely the same, so it would make sense.

**Comment by delvh** [comment]
Mostly.
While working on it, I've already noticed one inconsistency between org and user explore pages that borders on buggy behavior that I'll fix at the same time.

**Comment by delvh** [comment]
Correction: Two bugs.

**Comment by delvh** [comment]
@lunny moving the discussion here.

How can it be more confusing to have both in the same place?
It is completely confusing as it is **now**: I need to know if I'm looking for a user or an organization which I often don't know when looking at the explore page.
I agree, we can improve the display to make it more obvious who is a user, and who is an organization.
Still, I think it is plain wrong to separate them as we do at the moment.
We can add things like a little icon whether someone is a person or an organization, or filter by `users/orgs only`, but on the same page.

Additionally, I'd like to keep the refactoring clean, so that we add in these enhancements afterward.
What do you think?

And after all, `/explore/users` already behaves differently than `/explore/organizations` which is making me crazy.
See the PR description for what I mean.

**Comment by silverwind** [comment]
I tend to agree on the part that users and orgs are mostly treated the same. They both are in the same primary key. It's a similar situation with issues and pulls, they share the same primary key as well. Many places in Gitea treat issues and pulls the same, so I think it makes sense to clean this up for user/org as well.

**Comment by lunny** [comment]
Yes, I think your example is an evidence that I'm right. We have issues and pulls two different UI even though they shared the part of database. Only fewer places we will mix them.(Only milestones if I'm not wrong)

I still think they are different and we may encounter unresolvable conflicts after merging the two pages. This will become a limitation about how this page is being improved.
